mscr-ui/src/common/components/schema-and-crosswalk-actionmenu/index.tsx

```diff
@@ -0,0 +1,360 @@
+import { Metadata } from '@app/common/interfaces/metadata.interface';
+import {
+  useDeleteCrosswalkMutation,
+  usePatchCrosswalkMutation,
+} from '@app/common/components/crosswalk/crosswalk.slice';
+import {
+  useDeleteSchemaMutation,
+  usePatchSchemaMutation,
+} from '@app/common/components/schema/schema.slice';
+import { useTranslation } from 'next-i18next';
+import { ActionMenu, ActionMenuItem } from 'suomifi-ui-components';
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { ActionMenuTypes } from '@app/common/interfaces/search.interface';
+import { State } from '@app/common/interfaces/state.interface';
+import ConfirmModal from '@app/common/components/confirmation-modal';
+import { useStoreDispatch } from '@app/store';
+import {
+  clearNotification,
+  setNotification,
+} from '@app/common/components/notifications/notifications.slice';
+import { mscrSearchApi } from '@app/common/components/mscr-search/mscr-search.slice';
+import { CrosswalkWithVersionInfo } from '@app/common/interfaces/crosswalk.interface';
+import { SchemaWithVersionInfo } from '@app/common/interfaces/schema.interface';
+
+interface SchemaAndCrosswalkActionmenuProps {
+  type: ActionMenuTypes;
+  metadata: SchemaWithVersionInfo | CrosswalkWithVersionInfo | Metadata;
+  isMappingsEditModeActive: boolean;
+  refetchMetadata: () => void;
+  buttonCallbackFunction?: (action: string) => void;
+}
+
+export default function SchemaAndCrosswalkActionMenu({
+  type,
+  metadata,
+  isMappingsEditModeActive,
+  refetchMetadata,
+  buttonCallbackFunction = () => {
+    return;
+  },
+}: SchemaAndCrosswalkActionmenuProps) {
+  const { t } = useTranslation('common');
+  const dispatch = useStoreDispatch();
+  const [isEditModeActive, setIsEditModeActive] = useState(false);
+  const [patchCrosswalk] = usePatchCrosswalkMutation();
+  const [patchSchema] = usePatchSchemaMutation();
+  const [deleteSchema] = useDeleteSchemaMutation();
+  const [deleteCrosswalk] = useDeleteCrosswalkMutation();
+  const [isPublishConfirmModalOpen, setPublishConfirmModalOpen] =
+    useState(false);
+  const [isInvalidateConfirmModalOpen, setInvalidateConfirmModalOpen] =
+    useState(false);
+  const [isDeprecateConfirmModalOpen, setDeprecateConfirmModalOpen] =
+    useState(false);
+  const [isDeleteModalOpen, setDeleteModalOpen] = useState(false);
+  const [isRemoveModalOpen, setRemoveModalOpen] = useState(false);
+
+  const performModalAction = (action: string) => {
+    setPublishConfirmModalOpen(false);
+    setDeleteModalOpen(false);
+    setRemoveModalOpen(false);
+    setInvalidateConfirmModalOpen(false);
+    setDeprecateConfirmModalOpen(false);
+    if (action === 'close') {
+      return;
+    }
+    const payload: { versionLabel: string; state?: State } = {versionLabel: metadata.versionLabel};
```

Comment:
There's a bug in the backend, so temporarily have to pass versionLabel in addition to state